apps/api/src/shared/decorators/user.decorator.ts

```diff
@@ -0,0 +1,10 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+
+/**
+ * Retrieve the logged in user ID from the request context
+ */
+export const CurrentUser = createParamDecorator((data: unknown, context: ExecutionContext) => {
+  const httpContext = context.switchToHttp();
+  const req = httpContext.getRequest();
+  return req.user;
```

Comment:
where is the user property being injected?

Reply:
It's something Passport does [automatically](https://docs.nestjs.com/techniques/authentication#login-route).

"*Passport automatically creates a user object, based on the value we return from the validate() method, and assigns it to the Request object as req.user*"